drivers/sensor/apds9960/apds9960.c

```diff
@@ -0,0 +1,584 @@
+		return -EIO;
+	}
+
+	if (i2c_reg_write_byte(data->i2c, APDS9960_I2C_ADDRESS,
+		APDS9960_WTIME_REG, APDS9960_DEFAULT_WTIME) < 0) {
+		SYS_LOG_DBG("Defualt wait time not set ");
+		return -EIO;
+	}
+
+	if (i2c_reg_write_byte(data->i2c, APDS9960_I2C_ADDRESS,
+		APDS9960_PPULSE_REG, APDS9960_DEFAULT_PROX_PPULSE) < 0) {
+		SYS_LOG_DBG("Defualt wait time not set ");
+		return -EIO;
+	}
+
+	if (i2c_reg_write_byte(data->i2c, APDS9960_I2C_ADDRESS,
+		APDS9960_POFFSET_UR_REG, APDS9960_DEFAULT_POFFSET_UR) < 0) {
+		SYS_LOG_DBG("Defualt wait time not set ");
+		return -EIO;
+	}
+
+
+	if (i2c_reg_write_byte(data->i2c, APDS9960_I2C_ADDRESS,
+		APDS9960_POFFSET_DL_REG, APDS9960_DEFAULT_POFFSET_DL) < 0) {
+		SYS_LOG_DBG("Defualt wait time not set ");
+		return -EIO;
+	}
+
+	if (i2c_reg_write_byte(data->i2c, APDS9960_I2C_ADDRESS,
+		APDS9960_CONFIG1_REG, APDS9960_DEFAULT_CONFIG1) < 0) {
+		SYS_LOG_DBG("Defualt wait time not set ");
+		return -EIO;
+	}
+
+	if (!apds9960_setleddrive(dev, APDS9960_DEFAULT_LDRIVE)) {
+		SYS_LOG_DBG("LEDdrive not set");
+		return -EIO;
+	}
+
+	if (!apds9960_setproximitygain(dev, APDS9960_DEFAULT_PGAIN)) {
+		SYS_LOG_DBG("proximity gain not set");
+		return -EIO;
+	}
+
+	if (!apds9960_setambientlightgain(dev, APDS9960_DEFAULT_AGAIN)) {
+		SYS_LOG_DBG("ambient light gain not set");
+		return -EIO;
+	}
+
+	if (!apds9960_setproxintlowthresh(dev, APDS9960_DEFAULT_PILT)) {
+		SYS_LOG_DBG("prox low threshold not set");
+		return -EIO;
+	}
+
+	if (!apds9960_setproxinthighthresh(dev, APDS9960_DEFAULT_PIHT)) {
+		SYS_LOG_DBG("prox high threshold not set");
+		return -EIO;
+	}
+
+	if (!apds9960_setlightintlowthresh(dev, APDS9960_DEFAULT_AILT)) {
+		SYS_LOG_DBG("light low threshold not set");
+		return -EIO;
+	}
+
+	if (!apds9960_setlightinthighthresh(dev, APDS9960_DEFAULT_AIHT)) {
+		SYS_LOG_DBG("light high threshold not set");
+		return -EIO;
+	}
+
+	if (i2c_reg_write_byte(data->i2c, APDS9960_I2C_ADDRESS,
+		APDS9960_PERS_REG, APDS9960_DEFAULT_PERS) < 0) {
+		SYS_LOG_DBG("ALS interrupt persistence not set ");
+		return -EIO;
+	}
+
+	if (i2c_reg_write_byte(data->i2c, APDS9960_I2C_ADDRESS,
+		APDS9960_CONFIG2_REG, APDS9960_DEFAULT_CONFIG2) < 0) {
+		SYS_LOG_DBG("clear diode saturation interrupt is not enabled");
+		return -EIO;
+	}
+
+	apds9960_proxy_setup(dev, gain);
+	apds9960_ambient_setup(dev, gain);
+
+	return 0;
+}
+
+static struct sensor_driver_api apds9960_driver_api = {
+	.sample_fetch = &apds9960_sample_fetch,
+	.channel_get = &apds9960_channel_get,
+};
+
+int apds9960_init(struct device *dev)
+{
+	struct apds9960_data *data = dev->driver_data;
+	int als_gain;
+
+	data->i2c = device_get_binding(CONFIG_APDS9960_I2C_DEV_NAME);
+
+	if (data->i2c == NULL) {
+		SYS_LOG_DBG("Failed to get pointer to %s device!",
+		CONFIG_APDS9960_I2C_DEV_NAME);
+		return -EINVAL;
+	}
+
+	data->sample_c = 0;
+	data->sample_r = 0;
+	data->sample_g = 0;
+	data->sample_b = 0;
+	data->pdata = 0;
+
+	apds9960_sensor_setup(dev, als_gain);
```

Comment:
If I initialize it to zero , checkpatch.pl threw error saying global variable is initialized